src/test/netbase_tests.cpp

```diff
@@ -400,4 +399,22 @@ BOOST_AUTO_TEST_CASE(netpermissions_test)
     BOOST_CHECK(std::find(strings.begin(), strings.end(), "mempool") != strings.end());
 }
 
+BOOST_AUTO_TEST_CASE(netbase_dont_resolve_strings_with_embedded_nul_characters)
+{
+    CNetAddr addr;
+    BOOST_CHECK(LookupHost(std::string("127.0.0.1", 9), addr, false));
+    BOOST_CHECK(!LookupHost(std::string("127.0.0.1\0", 10), addr, false));
+    BOOST_CHECK(!LookupHost(std::string("127.0.0.1\0example.com", 21), addr, false));
+    BOOST_CHECK(!LookupHost(std::string("127.0.0.1\0example.com\0", 22), addr, false));
```

Comment:
nit, could drop the middle nul.

Reply:
To me the middle `\0` seems like a useful case to check. Whats the benefit in dropping it?

Reply:
already checked above.

Reply:
You mean that the test for  `BOOST_CHECK(!LookupHost(std::string("127.0.0.1\0", 10), addr, false));` is the same as the test for `BOOST_CHECK(!LookupHost(std::string("127.0.0.1\0example.com", 21), addr, false));`?